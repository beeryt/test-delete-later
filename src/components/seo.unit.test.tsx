import React from 'react'
import { render, waitFor } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')

import SEO from './seo'

describe('SEO', () => {
  const defaultMetadata = {
    title: 'Default Title',
    description: 'Default Description',
    siteUrl: 'https://example.com'
  }

  beforeAll(() => {
    useStaticQuery.mockImplementation(() => {
      return { site: { siteMetadata: defaultMetadata } }
    })
  })

  beforeEach(() => jest.clearAllMocks())

  afterAll(() => useStaticQuery.mockReset())

  it('renders defaults correctly', async () => {
    render(<SEO />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() => expect(document.title).toBe(defaultMetadata.title))
    await waitFor(() =>
      expect(
        document.head
          .querySelector('meta[name=description]')
          ?.attributes.getNamedItem('content')?.value
      ).toBe(defaultMetadata.description)
    )
    await waitFor(() =>
      expect(document.head.querySelector('link[rel=canonical]')).toBeNull()
    )
    await waitFor(() => expect(document.head).toMatchSnapshot())
  })

  it('correctly sets title', async () => {
    const title = 'Custom Title'
    render(<SEO title={title} />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() => expect(document.title).toEqual(title))
    await waitFor(() => expect(document.head).toMatchSnapshot())
  })

  it('correctly sets description', async () => {
    const description = 'Custom Description'
    render(<SEO desc={description} />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() =>
      expect(
        document.head
          .querySelector('meta[name=description]')
          ?.attributes.getNamedItem('content')?.value
      ).toBe(description)
    )
    await waitFor(() => expect(document.head).toMatchSnapshot())
  })

  it('correctly sets canonical link', async () => {
    const location = {
      pathname: '/blog/one'
    }
    render(<SEO location={location} />)
    expect(document.body).toMatchSnapshot()
    await waitFor(() =>
      expect(
        document.head
          .querySelector('link[rel=canonical]')
          ?.attributes.getNamedItem('href')?.value
      ).toBe(defaultMetadata.siteUrl + location.pathname)
    )
    await waitFor(() => expect(document.head).toMatchSnapshot())
  })
})
