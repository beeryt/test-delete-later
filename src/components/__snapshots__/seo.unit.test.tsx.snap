// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO correctly sets canonical link 1`] = `
<body>
  <div />
</body>
`;

exports[`SEO correctly sets canonical link 2`] = `
<head>
  <title>
    Default Title
  </title>
  <link
    data-react-helmet="true"
    href="https://example.com/blog/one"
    rel="canonical"
  />
  <meta
    content="Default Description"
    data-react-helmet="true"
    name="description"
  />
</head>
`;

exports[`SEO correctly sets description 1`] = `
<body>
  <div />
</body>
`;

exports[`SEO correctly sets description 2`] = `
<head>
  <title>
    Default Title
  </title>
  <meta
    content="Custom Description"
    data-react-helmet="true"
    name="description"
  />
</head>
`;

exports[`SEO correctly sets title 1`] = `
<body>
  <div />
</body>
`;

exports[`SEO correctly sets title 2`] = `
<head>
  <title>
    Custom Title
  </title>
  <meta
    content="Default Description"
    data-react-helmet="true"
    name="description"
  />
</head>
`;

exports[`SEO renders defaults correctly 1`] = `
<body>
  <div />
</body>
`;

exports[`SEO renders defaults correctly 2`] = `
<head>
  <title>
    Default Title
  </title>
  <meta
    content="Default Description"
    data-react-helmet="true"
    name="description"
  />
</head>
`;
